### DISCUSSION TITLE
Question on new PID Wrapping Code

### DISCUSSION THREAD

**Discussion by mtappen** [BroncBotz3481/YAGSL-Example#131] [Q&A]
I haven't had access to our robot for a couple of days, so I have been going through commits.  I noticed that the new Swerve Module code changed PID wrapping from (-180, 180) to (0, 90).  If I program the teleop drive to drive at fixed (1,-1) velocity, I can definitely get optimize to produce -45 degree setpoints in simulation.  Is this an issue?

Should the PID Wrapping be (-90, 90)?

Thanks!

**Comment by thenetworkgrinch** [comment]
Honestly, I am not sure. Upon testing it hasnt shown to be an issue, which was shocking to me at the time.

**Comment by mtappen** [comment]
I was just testing on our robot and we were seeing odd behavior until I changed the wrapping minimum to -90.

**Comment by mtappen** [comment]
I've written a change that eliminates PID wrapping altogether.  Unfortunately, REVLib does not describe how PID wrapping works.

[I've made this change](https://github.com/MarshallTappen/YAGSL-Example/blob/70a0469fda7d77bc956cddf49694c8ab7950c556/src/main/java/swervelib/SwerveModule.java#L178) that uses Rotation2d to eliminate the need for using PID wrapping on the controller.  I won't be able to test it on-robot until mid-week, but if it works, will submit a PR.

**Comment by mtappen** [comment]
Closing this discussion because the behavior was fixed by commits that changed the update from the encoders and PID wrapping to [0, 180)
